spec_to_code.py

```diff
@@ -0,0 +1,1100 @@
+            if line.strip().endswith(':= "'):
+                line = line.rstrip() + '""'
+            else:
+                line = line.rstrip() + '"'
+        
+        # Fix incomplete variable declarations
+        if 'var ' in line and ' := ' in line and not line.endswith(';'):
+            if not line.strip().endswith('}'):
+                line = line.rstrip() + ';'
+        
+        fixed_lines.append(line)
+    
+    return '\n'.join(fixed_lines)
+
+def fix_incomplete_lean_code(code):
+    """Fix incomplete Lean code patterns."""
+    lines = code.split('\n')
+    fixed_lines = []
+    
+    for i, line in enumerate(lines):
+        # Fix incomplete sorry statements
+        if 'sorry' not in line and (line.strip().startswith('theorem ') or 
+                                   line.strip().startswith('lemma ') or 
+                                   line.strip().startswith('def ')):
+            # Look ahead to see if there's a body
+            has_body = False
+            for j in range(i + 1, len(lines)):
+                if ':=' in lines[j] or 'sorry' in lines[j] or 'where' in lines[j]:
+                    has_body = True
+                    break
+                if (lines[j].strip().startswith('theorem ') or 
+                    lines[j].strip().startswith('lemma ') or 
+                    lines[j].strip().startswith('def ')):
+                    break
+            if not has_body and ':' in line:
+                # Add sorry for incomplete theorems/lemmas
+                fixed_lines.append(line)
+                fixed_lines.append('  sorry')
+                continue
+        
+        fixed_lines.append(line)
+    
+    return '\n'.join(fixed_lines)
+
+def detect_compilation_errors(output):
+    """Detect if the output contains compilation errors."""
+    return any(indicator in output.lower() for indicator in COMMON_COMPILATION_ERRORS)
+
+def check_tool_availability():
+    """Check if the language tool is available at the specified path."""
+    tool_path = get_tool_path()
+    try:
+        # Check if the tool executable exists
+        if not os.path.isfile(tool_path):
+            return False, f"{LANGUAGE_CONFIG.name} executable not found at: {tool_path}"
+        
+        # Check if the file is executable
+        if not os.access(tool_path, os.X_OK):
+            return False, f"{LANGUAGE_CONFIG.name} executable is not executable: {tool_path}"
+        
+        # Try to run tool with --help to verify it works
+        result = subprocess.run([tool_path, "--help"], 
+                              capture_output=True, text=True, timeout=10)
+        
+        if result.returncode != 0 and CURRENT_LANGUAGE != 'lean':
+            # Lean might not have --help, so we skip this check for Lean
+            return False, f"{LANGUAGE_CONFIG.name} executable failed to run: {result.stderr}"
+        
+        return True, f"{LANGUAGE_CONFIG.name} is available and working"
```

Comment:
i would rather wrap the return type in a small dataclass/namedtuple, but it's up to you @astefano